### PR TITLE
[stable/kong] No unnecessary ServiceAccount creation when IngressController and podSecurityPolicy are disabled

### LINKED ISSUE #19100: [stable/kong] ServiceAccount issues when upgrading to 0.27.0 chart
**Describe the bug**
When the previous helm chart is already used, upgrading to use the latest 0.27.0 chart errors out with ServiceAccount issues.

**Version of Helm and Kubernetes**:
Helm 2.16
K9s 1.14

**Which chart**:
stable/kong

**What happened**:
I have `ingressController.enabled: false` 
Jobs and Deployment now use the `ServiceAccount`.
The pre-migration job fails with:
```
48s         Warning   FailedCreate   job/dev-kong-kong-pre-upgrade-migrations   Error creating: pods "dev-kong-kong-pre-upgrade-migrations-" is forbidden: error looking up service account kong/dev-kong-kong: serviceaccount "dev-kong-kong" not found
```

The service account isn't created before the new job is executed, it seems to be a dependency in helm issue.

Even if changing `ingressController.serviceAccount.name: default` fails with the message:
```
kind ServiceAccount with the name "default" already exists in the cluster and wasn't defined in the previous release. Before upgrading, please either delete the resource from the cluster or remove it from the chart
```

Are there any known workarounds please ?

### PR DESCRIPTION
**Pull request by elogeart**
#### Is this a new chart
No

#### What this PR does / why we need it:
This PR stops the creation of a dedicated ServiceAccount if podSecurityPolicy is disabled and if Kong is not used as an Ingress Controller, as it is not necessary in this case.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #19100 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by bluemalkin** [comment]
@elogeart thanks for the PR fix ! Look forward to getting this released.

**Comment by rainest** [comment]
More background than comment on this PR. This is more or less fine as-is if you don't intend to use either of the features that require the account, but this issue still exists if you wish to later enable one:

07872e71581d003a90d0c78be8ef456b37e132dc previously removed most of the conditions around whether to use the ServiceAccount, as well as the conditions on creating it. This creates a circular dependency where the pre-upgrade migrations Job requires the ServiceAccount, but does not have access to it/fails, since it runs before any new template additions are applied.

Ideally, we'd like to be able to create the account if needed: simply disabling it and adding conditions around its use avoids the issue so long as it is disabled, but if you attempt to enable PSP after and then upgrade, you'll hit the same dependency loop.

I'm not sure offhand how to best accomplish this. Some halfway solutions that don't really work:
* Create an intermediate version that adds the account before requiring its use in pre-upgrade. This avoids the loop, but requires multi-stage chart version upgrades, and always creates the account, even if you have no intention of using it in the future.
* Add hook annotations to the account itself (not sure if this is doable or if hooks are only supported for some resource types). This is non-ideal since the result won't be managed by the release.

@pipo02mix do you have any thoughts on how the SA can be handled if it wasn't created during the initial install?

**Comment by rainest** [comment]
@elogeart could you integrate https://github.com/rainest/charts/commit/56847f2b4191d9df3b6fa23dc9411aff0a7ac507 into your branch? For some reason, GitHub isn't letting me open a PR to your fork.

Based on discussion with the community, it looks like creating and then deleting the ServiceAccount via its own pre-upgrade hook allows the migrations Job to proceed. I'd thought that it'd get deleted early and/or conflict with the standard user on subsequent upgrades, but it looks like it's handled correctly. 

1. Perform a fresh install with the controller and PSP disabled.
2. Attempt an upgrade that enables PSP using the existing code in this PR, to confirm that the pre-upgrade Job blocks on a missing ServiceAccount.
3. Add my commit, upgrade again to confirm it clears the block.
4. Perform an upgrade with an unrelated change, to confirm that the pre-upgrade is not now blocked on attempting to create a ServiceAccount that already exists.

Because (4) was not an issue, I am not using a different name for the temporary account.

File: stable/kong/Chart.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ maintainers:
 name: kong
 sources:
 - https://github.com/Kong/kong
-version: 0.27.0
+version: 0.28.1
```
Comment (hbagdi) [inline stable/kong/Chart.yaml]: This should be 0.27.2.
Please also rebase on the latest master.

**Comment by hbagdi** [comment]
You also need to write a changelog entry in the README file.
Please see Travis's commit and incorporate it here and we will be good to merge. Thank you for the fix @elogeart

**Review by hbagdi — APPROVED** [review]
/lgtm

**Comment by pipo02mix** [comment]
thanks for the fix @rainest and @elogeart I did not notice the issue so far.

**Pull request merged**